From 7a464270d33423cab74ebd7fc65de13388fb6e06 Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Thu, 2 Apr 2015 15:26:41 +0800
Subject: [PATCH 356/360] atomisp2: Clear update flags after parameters are set

Change-Id: I61f770d5122c4daecbe04c1f2ad6b623c05dcd18
Fixes-Issue: #JIMINAN-28313
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/49438
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Build: IMIN-legacy (SC)
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c      | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index d0175c6..677daa0 100755
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -1733,6 +1733,8 @@ irqreturn_t atomisp_isr_thread(int irq, void *isp_ptr)
 					&asd->params.css_param.dz_config);
 			atomisp_css_update_isp_params(asd);
 			asd->params.css_update_params_needed = false;
+			memset(&asd->params.css_param.update_flag, 0,
+			       sizeof(struct atomisp_parameters));
 			frame_done_found[asd->index] = false;
 		}
 		atomisp_setup_flash(asd);
-- 
1.9.1

